assets/translate.js

```diff
@@ -127,8 +132,11 @@ $( window ).on( 'load', function () {
 				// Go through all fields and construct the request parameters.
 				$( '.translation-fields .oo-ui-fieldLayout' ).each( function () {
 					var fieldLayout = OO.ui.infuse( $( this ) ),
-						tspanId = fieldLayout.getField().data[ 'tspan-id' ];
-					requestParams[ tspanId ] = fieldLayout.getField().getValue();
+						tspanId = fieldLayout.getField().data[ 'tspan-id' ],
+						text = fieldLayout.getField().getValue(),
+						textChanged = text !== '' && text !== appConfig.translations[ tspanId ][ targetLangWidget.getValue() ];
```

Comment:
The text of the stored translation is in the `text` property, i.e. this should be `appConfig.translations[ tspanId ][ targetLangWidget.getValue() ].text`.